MapboxNavigation/NavigationViewController.swift

```diff
@@ -75,12 +75,19 @@ public protocol NavigationViewControllerDelegate {
      
      If this method is unimplemented, the navigation map view draws the route line using an `MGLLineStyleLayer`.
      */
-    @objc optional func navigationMapView(_ mapView: NavigationMapView, routeStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer?
+    @objc optional func navigationMapView(_ mapView: NavigationMapView, routeWaypointCircleStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer?
+    
+    /**
+     Returns an `MGLStyleLayer` that determines the appearance of the route line.
+     
+     If this method is unimplemented, the navigation map view draws the route line using an `MGLLineStyleLayer`.
+     */
+    @objc optional func navigationMapView(_ mapView: NavigationMapView, routeWaypointSymbolStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer?
     
```

Comment:
What if the waypoint needs to be represented by an annotation, to match the main map in their application? There should be alternative, optional methods that mirror `MGLMapViewDelegate.mapView(_:imageForAnnotation:)` and `MGLMapViewDelegate.mapView(_:viewForAnnotation:)`. When either `navigationMapView(_:imageForAnnotation:waypoint:)` or `navigationMapView(_:viewForAnnotation:waypoint:)` is implemented, NavigationMapView would take care of adding an MGLPointAnnotation instead of a source and style layer.

/ref #114

Reply:
#1993